src/inference/src/dev/plugin_config.cpp

```diff
@@ -0,0 +1,319 @@
+
+#ifdef ENABLE_DEBUG_CAPS
+    apply_env_options();
+#endif
+
+    // Clear properties after finalize_impl to be able to check if a property was set by user during plugin-side finalization
+    m_user_properties.clear();
+
+    m_is_finalized = true;
+}
+
+bool PluginConfig::visit_attributes(ov::AttributeVisitor& visitor) {
+    visitor.on_attribute("m_user_properties", m_user_properties);
+    for (auto& prop : m_options_map) {
+        visitor.on_attribute(prop.first + "__internal", prop.second);
+    }
+
+    return true;
+}
+
+void PluginConfig::apply_env_options() {
+    ov::AnyMap env_properties = read_env();
+    cleanup_unsupported(env_properties);
+    for (auto& [name, val] : env_properties) {
+        std::cout << "Non default env value for " << name << " = " << val.as<std::string>() << std::endl;
+    }
+    set_property(env_properties);
+}
+
+void PluginConfig::apply_config_options(std::string_view device_name, std::string_view config_path) {
+    if (!config_path.empty()) {
+        ov::AnyMap config_properties = read_config_file(std::string(config_path), device_name);
+        cleanup_unsupported(config_properties);
+#ifdef ENABLE_DEBUG_CAPS
+        for (auto& [name, val] : config_properties) {
+            std::cout << "Non default config value for " << name << " = " << val.as<std::string>() << std::endl;
+        }
+#endif
+        set_property(config_properties);
+    }
+}
+
+ov::AnyMap PluginConfig::read_config_file(std::string_view filename, std::string_view target_device_name) const {
+    if (filename.empty())
+        return {};
+
+    ov::AnyMap config;
+
+    std::ifstream ifs(std::string{filename});
+    if (!ifs.is_open()) {
+        return config;
+    }
+
+    nlohmann::json json_config;
+    try {
+        ifs >> json_config;
+    } catch (const std::exception&) {
+        return config;
+    }
+
+    DeviceIDParser parser(std::string{target_device_name});
+    for (auto item = json_config.cbegin(), end = json_config.cend(); item != end; ++item) {
+        const std::string& device_name = item.key();
+        if (DeviceIDParser(device_name).get_device_name() != parser.get_device_name())
+            continue;
+
+        const auto& item_value = item.value();
+        for (auto option = item_value.cbegin(), item_value_end = item_value.cend(); option != item_value_end; ++option) {
+            config[option.key()] = option.value().get<std::string>();
+        }
+    }
+
+    return config;
+}
+
+ov::Any PluginConfig::read_env(const std::string& option_name, std::string_view prefix, const ConfigOptionBase* option) {
+    auto var_name = std::string(prefix) + option_name;
+    const auto& val = ov::util::getenv_string(var_name.c_str());
+
+    if (!val.empty()) {
+        if (dynamic_cast<const TypedOption<bool>*>(option) != nullptr) {
+            constexpr std::array<std::string_view, 4> off = {"0", "false", "off", "no"};
+            constexpr std::array<std::string_view, 4> on = {"1", "true", "on", "yes"};
```

Comment:
Can be ov::Any re-sused to read this string as bool ? to avoid this strings duplication or have them in one place ?

Reply:
Yeah, since I extended ov::Any's string parser this code is not needed anymore.